packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -66,4 +66,11 @@ abstract class BinaryMessenger {
 /// keeps track of which handlers have been registered on each channel so
 /// it may dispatch incoming messages to the registered handler.
 @Deprecated('Use ServicesBinding.instance.defaultBinaryMessenger instead.')
-BinaryMessenger get defaultBinaryMessenger => ServicesBinding.instance.defaultBinaryMessenger;
+BinaryMessenger get defaultBinaryMessenger {
+  const String errMsg =
+      'ServicesBinding is not properly initialized when accessing the [defaultBinaryMessenger] instance. '
+      'If you need the binding to be constructed before calling [runApp], e.g. using platform channels/plugins, '
+      'call the `WidgetsFlutterBinding.ensureInitialized()` function.';
+  assert(ServicesBinding.instance != null, errMsg);
```

Comment:
Better to move this assert to the `ServicesBinding.instance` getter, so it'll catch all the places where we're now calling `ServicesBinding.instance.defaultBinaryMessenger` directly in the framework where we were previously calling `defaultBinaryMessenger`

Reply:
Ya.. thought about this, but wondering whether that would be too aggressive?
This means callers cannot do something like:

if (ServicesBinding.instance == null) {
  // Do some initialization
  WidgetsFlutterBinding.ensureInitialization();
}

In addition, I did not find simple patterns for other bindings.

Reply:
That's a good point.  But there are a number of places in this PR where we've changed from calling `defaultBinaryMessenger` to calling `ServicesBinding.instance.defaultBinaryMessenger`, and those calls sites could be affected.  If you update those call sites to stay with their existing call of `defaultBinaryMessenger`, then you'll be covered (you'll have to suppress the deprecation warning in those call sites as well for now).

Reply:
That sounds good and it will help make the error message more obvious during migration. I'll replace `ServicesBinding.instance.defaultBinaryMessenger` with `defaultBinaryMessenger `, except tests (tests are more self-contained and should be fine).

Reply:
Done. PTAL.